modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/recovery/ItPartitionStatesTest.java

```diff
@@ -0,0 +1,211 @@
+
+        assertErrOutputContains("Some distribution zones are missing: [UNKNOWN_ZONE]");
+
+        assertOutputIsEmpty();
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testPartitionStatesMissingPartition(boolean global) {
+        String unknownPartition = "-1";
+
+        execute("recovery", "partition-states",
+                CLUSTER_URL_OPTION, NODE_URL,
+                RECOVERY_PARTITION_IDS_OPTION, unknownPartition,
+                global ? RECOVERY_PARTITION_GLOBAL_OPTION : RECOVERY_LOCAL_OPTION);
+
+        assertErrOutputContains("Some partitions are missing: [-1]");
+
+        assertOutputIsEmpty();
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testPartitionStatesMissingNode() {
+        String unknownNode = "unknown_node";
+
+        execute("recovery", "partition-states",
+                CLUSTER_URL_OPTION, NODE_URL,
+                RECOVERY_NODE_NAMES_OPTION, unknownNode,
+                RECOVERY_LOCAL_OPTION);
+
+        assertErrOutputContains("Some nodes are missing: [unknown_node]");
+
+        assertOutputIsEmpty();
+    }
+
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testPartitionStatesEmptyResult(boolean global) {
+        execute("recovery", "partition-states",
+                CLUSTER_URL_OPTION, NODE_URL,
+                RECOVERY_ZONE_NAMES_OPTION, "empty_zone",
+                global ? RECOVERY_PARTITION_GLOBAL_OPTION : RECOVERY_LOCAL_OPTION
+        );
+
+        checkOutput(global, Set.of(), Set.of());
+    }
+
+    private void checkOutput(boolean global, Set<String> zoneNames, Set<String> nodes) {
+        assertErrOutputIsEmpty();
+
+        if (global) {
+            assertOutputDoesNotContain("Node name");
+        } else {
+            assertOutputContains("Node name");
+
+            if (!nodes.isEmpty()) {
+                assertOutputContainsAnyIgnoringCase(nodes);
+            }
+
+            Set<String> anotherNodes = CollectionUtils.difference(nodeNames, nodes);
+
+            if (!anotherNodes.isEmpty()) {
+                assertOutputDoesNotContainIgnoreCase(anotherNodes);
+            }
+        }
+
+        assertOutputContains("Zone name");
+        assertOutputContains("Table name");
+        assertOutputContains("Partition ID");
+        assertOutputContains("State");
+
+        if (!zoneNames.isEmpty()) {
+            assertOutputContainsAllIgnoringCase(zoneNames);
+
+            Set<String> tableNames = zoneNames.stream().map(it -> it + "_table").collect(toSet());
+
+            assertOutputContainsAllIgnoringCase(tableNames);
+        }
+
+        Set<String> anotherZones = CollectionUtils.difference(ZONES, zoneNames);
+
+        if (!anotherZones.isEmpty()) {
+            assertOutputDoesNotContainIgnoreCase(anotherZones);
+        }
+
+        if (!zoneNames.isEmpty() && nodeNames.isEmpty()) {
+            assertOutputContainsAnyIgnoringCase(STATES);
+        }
+    }
```

Comment:
Is there a test for a situation when we specify partition IDs explicitly (that would check that we only get results for the requested partition IDs)?

Same for filtering on node names